webrender_api/src/display_list.rs

```diff
@@ -594,6 +602,20 @@ impl DisplayListBuilder {
         }
     }
 
+    pub fn save(&self) -> DisplayListBuilderSaveState {
+        DisplayListBuilderSaveState {
+            clip_stack_len: self.clip_stack.len(),
```

Comment:
wait, you are saving the clip stack length but not the actual contents?

Reply:
Yes, it will only "restore" by removing clips that were pushed since the save. This is sufficient for all actual use-cases, I think? Having trouble picturing a case where you'd save and then pop something that was pushed before the save.

Reply:
I don't understand. This is a stack, not a growing-only vector. We don't know what have been pushed/popped from the stack in the meantime. Having the stack containing newer items in place of the older ones after resetting would open a whole can of worms, sounds very unsafe to me.